consensus/poet/core/sawtooth_poet/poet_consensus/utils.py

```diff
@@ -113,3 +118,155 @@ def build_certificate_list(block_header,
         LOGGER.error('Error getting block: %s', ke)
 
     return list(certificates)
+
+
+def create_validator_state(validator_info, current_validator_state):
+    """Starting with the current validator state (or None), create validator
+     validator state for the validator.
+
+    Args:
+        validator_info (ValidatorInfo): Information about the validator
+        current_validator_state (ValidatorState): The current validator
+            state we are going to update or None if there isn't any and
+            we are to create default validator state
+
+    Returns:
+        ValidatorState object
+    """
+
+    # Start with the default validator state (i.e., assuming that none existed
+    # to start with) and update accordingly.  If it does exist and the PoET
+    # public key is the same, then we are just going to update counts.  If it
+    # does exist but the PoET public key has changed, we need to update
+    # statistics and take note of the new key.
+    key_block_claim_count = 1
+    poet_public_key = validator_info.signup_info.poet_public_key
+    total_block_claim_count = 1
+
+    if current_validator_state is not None:
+        total_block_claim_count = \
+            current_validator_state.total_block_claim_count + 1
+
+        if validator_info.signup_info.poet_public_key == \
+                current_validator_state.poet_public_key:
+            key_block_claim_count = \
+                current_validator_state.key_block_claim_count + 1
```

Comment:
Does this need to be computed over a window for z-test? i.e. of the last 10 blocks this poet public key claimed 5 blocks. (but over the lifetime of that key it claimed 500 blocks)